ERCS/erc-7629.md

```diff
@@ -0,0 +1,296 @@
+- Defines essential functions and events to support token interactions, conversions, and queries.
+- Implements a low gas consumption [ERC-20](./eip-20.md) mode to maintain gas efficiency comparable to typical [ERC-20](./eip-20.md) token transfers.
+
+Compliant contracts MUST implement the following Solidity interface:
+
+```solidity
+
+pragma solidity ^0.8.0;
+/**
+ * @title ERC-7629 Unify Token Interface
+ * @dev This interface defines the ERC-7629 Unify Token, which unifies ERC-721 and ERC-20 assets.
+ */
+interface IERC7629 {
+    // ERC-20 Transfer event
+    event ERC20Transfer(
+        address indexed from,
+        address indexed to,
+        uint256 amount
+    );
+
+    // ERC-721 Transfer event
+    event ERC721Transfer(
+        address indexed from,
+        address indexed to,
+        uint256 indexed tokenId
+    );
+
+    // Approval event for ERC-20 and ERC-721
+    event Approval(
+        address indexed owner,
+        address indexed approved,
+        uint256 indexed tokenId
+    );
+
+    // ApprovalForAll event for ERC-721
+    event ApprovalForAll(
+        address indexed owner,
+        address indexed operator,
+        bool approved
+    );
+
+    // ERC-20 to ERC-721 Conversion event
+    event ERC20ToERC721(address indexed to, uint256 amount, uint256[] tokenIds);
+
+    // ERC-721 to ERC-20 Conversion event
+    event ERC721ToERC20(address indexed to, uint256 tokenId, uint256 amount);
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() external view returns (string memory);
+
+    /**
+     * @dev Returns the symbol of the token.
+     */
+    function symbol() external view returns (string memory);
+
+    /**
+     * @dev Returns the number of decimals used in the token.
+     */
+    function decimals() external view returns (uint8);
+
+    /**
+     * @dev Returns the total supply of the ERC-20 tokens.
+     */
+    function totalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-20 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function balanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of ERC-20 tokens.
+     */
+    function erc20TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-20 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc20BalanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of ERC-721 tokens.
+     */
+    function erc721TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-721 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc721BalanceOf(address owner) external view returns (uint256);
+
```

Comment:
```suggestion
    /**
     * @notice Get the approved address for a single NFT
     * @dev Throws if `tokenId` is not a valid NFT.
     * @param tokenId The NFT to find the approved address for
     * @return The approved address for this NFT, or the zero address if there is none
     */
    function getApproved(uint256 tokenId) external view returns (address);

```
A `getApproved` function is necessary to determine if an NFT has been authorized for a specific address, while this ERC only incorporates the `isApprovedForAll` function.